File: packages/hmssdk_flutter/ios/Classes/Models/polls/HMSPollResponsePeerInfoExtension.swift
//
//  HMSPollResponsePeerInfoExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 20/02/24.
//

import Foundation
import HMSSDK

class HMSPollResponsePeerInfoExtension {

    static func toDictionary(hmsPollResponsePeerInfo: HMSPollResponsePeerInfo?) -> [String: Any?]? {

        if let hmsPollResponsePeerInfo {

            var map = [String: Any?]()

            map["hash"] = hmsPollResponsePeerInfo.userHash
            map["peer_id"] = hmsPollResponsePeerInfo.peerID
            map["user_id"] = hmsPollResponsePeerInfo.customerUserID
            map["username"] = hmsPollResponsePeerInfo.userName

            return map
        } else {
            return nil
        }

    }

}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/poll_extension/HMSPollResponsePeerInfoExtension.kt
package live.hms.hmssdk_flutter.poll_extension

import live.hms.video.polls.models.network.HMSPollResponsePeerInfo

class HMSPollResponsePeerInfoExtension {
    companion object {
        fun toDictionary(hmsPollResponsePeerInfo: HMSPollResponsePeerInfo?): HashMap<String, Any?>? {
            if (hmsPollResponsePeerInfo == null) {
                return null
            }

            val map = HashMap<String, Any?>()

            map["hash"] = hmsPollResponsePeerInfo.hash
            map["peer_id"] = hmsPollResponsePeerInfo.peerid
            map["user_id"] = hmsPollResponsePeerInfo.userid
            map["username"] = hmsPollResponsePeerInfo.username

            return map
        }
    }
}
